hazelcast-client-new/src/main/java/com/hazelcast/client/impl/ClientServiceFactory.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.client.connection.ClientConnectionManager;
+
+public interface ClientServiceFactory {
+
+    ClientConnectionManager createConnectionManager(ClientConfig config, HazelcastClientInstanceImpl client);
```

Comment:
I started implementing thinking I am gonna need to mock more services. But it turned out connection manager is enough. I am not sure if we are gonna need more later. I will change to ClientConnectionManagerFactory for now. I can always rename when we had to add more services. Or we can add more factories.